### PR TITLE
DOP-4242: Use release, not tag push for CI/CD

### PR DESCRIPTION
**Pull request by mayaraman19**
[DOP-4242](https://jira.mongodb.org/browse/DOP-4242)

This PR changes all instances of `on.push.tag` to `on.release` so that the CI/CD workflow is only activated upon release. please ignore the commit history :sob:

To test, I've created [this dummy repository](https://github.com/mayaraman19/actions-test) which has actions that are run only upon release. It seems to be working - feel free to clone, contribute, release, and play around with it.

### CODE REVIEW

**Review by seungpark — CHANGES_REQUESTED** [review]
This will be a good opportunity to update the readme for releases! 🙏

File: .github/workflows/deploy-prd-ecs.yml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 on:
-  push:
-    tags:
-      - 'v*'
+  release:
+    types: [released]
```
[the remark this answers is not in the document]
Comment (mayaraman19) [reply .github/workflows/deploy-prd-ecs.yml:3]: I talked about this with Cassidy and he told me to use `released`, but I'll confirm with him!

[the remark this answers is not in the document]
Comment (casthewiz) [reply .github/workflows/deploy-prd-ecs.yml:3]: @seungpark We don't want to catch drafts as a part of this action - I want our releases to be highly intentional and as tightly scoped as possible 👍

[the remark this answers is not in the document]
Comment (seungpark) [reply .github/workflows/deploy-prd-ecs.yml:3]: @casthewiz not creation of a draft, but transitioning a draft to published will not be caught within `released` types as in the first link above.

if this is still desirable, @mayaraman19 we should add to the [base readme of steps to release](https://github.com/mongodb/docs-worker-pool/blob/master/README.md#release-process). will pushing a tag fire a release? - i'm not sure of this. or do we need to use `gh release`

[the remark this answers is not in the document]
Comment (mayaraman19) [reply .github/workflows/deploy-prd-ecs.yml:3]: @seungpark based on testing with my dummy repository linked in the comment, I think that transitioning from draft to published should actually trigger the CI/CD workflow. 

And yes, I should update the base readme. Pushing a tag should not fire a release.

[the remark this answers is not in the document]
Comment (seungpark) [reply .github/workflows/deploy-prd-ecs.yml:3]: !! thanks for the test repo. forked and tested! https://github.com/seungpark/actions-test/actions. LGTM. i have seen some SO posts about the ambiguity of `released` vs `published` in the docs....

tested with GH UI that draft -> published does in fact fire this

File: README.md [hunk 2]
```diff
@@ -157,7 +157,8 @@ docs-worker-pool contains various triggers for release to higher environments. C
 
 - Create release tags. We currently follow [semver](https://semver.org/) standards.
 - If you don't have push access, open an issue or otherwise contact a contributor with administrator privileges.
-- Verify that the deploy-production-ec2 workflow executed successfully for both job runs across both production instances.
+- Create a release associated with the tag using the GitHub UI or by running `gh release create`.
```
Comment (seungpark) [inline README.md:160]: might be an extra space at end of line, based [on error](https://github.com/mongodb/docs-worker-pool/actions/runs/7631054222/job/20788127472?pr=968)

File: .github/workflows/README.md [hunk 3]
```diff
@@ -2,6 +2,14 @@
 
 This README describes the various workflows defined for the docs-worker-pool repository.
 
+## Releases
+
+The release process occurs whenever someone releases a specific tag via the GitHub UI. Three separate workflows are run upon release:
```
Comment (seungpark) [inline .github/workflows/README.md:7]: minor nitpick. not specific to the Github UI

**Review by seungpark — APPROVED** [review]
LGTM. minor notes below!

**Pull request merged**